locality-checker/pkg/kubernetes/locality.go

```diff
@@ -0,0 +1,122 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type LocalityChecker struct {
+	// The name of the Kubernetes node the container is running on.
+	NodeName string
+
+	// The directory to write locality information.
+	WritePath string
+
+	// Whether to error if node does not have region and zone labels.
+	ErrorOnMissingLabels bool
+}
+
+type localityInfo struct {
+	Region string
+	Zone   string
+}
+
+func (l *LocalityChecker) WriteLocality(ctx context.Context) error {
+	localityInfo, err := l.getLocalityInfo(ctx)
+	if err != nil {
+		return err
+	}
+	if localityInfo == nil {
+		return nil
+	}
+	return l.writeLocalityInfo(ctx, localityInfo)
+}
+
+func (l *LocalityChecker) getLocalityInfo(ctx context.Context) (*localityInfo, error) {
+	labels, err := l.getNodeLabels(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting node labels failed")
+	}
+	region, err := l.getRegion(labels)
+	if err != nil {
+		if !l.ErrorOnMissingLabels {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "no region labels found")
+	}
+	zone, err := l.getZone(labels)
+	if err != nil {
+		if !l.ErrorOnMissingLabels {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "no zone labels found")
+	}
+	return &localityInfo{
+		Region: region,
+		Zone:   zone,
+	}, nil
+}
+
+func (l *LocalityChecker) writeLocalityInfo(ctx context.Context, localityInfo *localityInfo) error {
+	err := l.writeFile("region", localityInfo.Region)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
It may be better practice to return these errors all the way up to `main.go` and then call `log.Fatal` there